sympy/polys/polyoptions.py

```diff
@@ -283,7 +284,7 @@ def default(cls):
     def preprocess(cls, gens):
         if isinstance(gens, Basic):
             gens = (gens,)
-        elif len(gens) == 1 and hasattr(gens[0], '__iter__'):
+        elif len(gens) == 1 and iterable(gens[0]):
```

Comment:
The code here won't actually work with an iterator. Probably this should check `is_sequence` rather than `iterable`.

Reply:
It should work. len is called on `gens` while `gens[0]` should be a generator. I can change it to `is_sequence`

Reply:
It's the subsequent code that iterates over gens multiple times that won't work.

Reply:
Ok, i'll change it.